pipeline/runtime/filterfileutil.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Harness Inc. All rights reserved.
+// Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+// that can be found in the licenses directory at the root of this repository, also available at
+// https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+package runtime
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/harness/lite-engine/internal/filesystem"
+	"github.com/harness/ti-client/types"
+)
+
+func populateItemInFilterFile(selectTestResp types.SelectTestsResp, filePath string, fs filesystem.FileSystem, isFilterFilePresent bool) error {
+
+	if !isFilterFilePresent { // If filter file is not present then we will simply run all the test cases
+		log.Println("We are here where no filter file is being created") //Revert
+		return nil
+	}
+
+	log.Println("Attempting to create filter file")
+	testResp := selectTestResp.Tests
+	filterFile := fmt.Sprintf("%s/filter", filePath)
+	f, err := fs.Create(filterFile)
+	if err != nil {
+		log.Println(fmt.Sprintf("could not create file %s", filterFile), err)
+		return err
+	}
+
+	var data string
+	for i := range testResp {
+		test := testResp[i]
```

Comment:
What if the response is nil, will there be empty or no filter file. How does it effect the behaviour of test selection

Reply:
If there is no filter file we are simply returning around Line no 20. If filterfile is present but null then empty data will be written. Rest Java agent will take care on the basis of contents of filter file